### PR TITLE
Adds gitlab docs

### PR DESCRIPTION
**Pull request by orta**
I've not run through this myself, but it looks about right

### CODE REVIEW

**Comment by f-meloni** [comment]
I think I should do the same for Danger-Swift

**Comment by orta** [comment]
👍

File: docs/usage/gitlab.html.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+The DSL is expansive, you can see all the details inside the [Danger JS Reference][ref], but the TLDR is:
+
+```ts
+danger.gitlab.
+  /** The pull request and repository metadata */
+  metadata: RepoMetaData
+  /** The Merge Request metadata */
+  mr: BitBucketCloudPRDSL
```
Comment (sluongng) [inline docs/usage/gitlab.html.md]: Why was BitBucket referenced here?

File: docs/usage/gitlab.html.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  /** The commits associated with the merge request */
+  commits: BitBucketCloudCommit[]
```
Comment (sluongng) [inline docs/usage/gitlab.html.md]: BitBucket should not be mentioned here?

File: docs/usage/gitlab.html.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+- `DANGER_GITLAB_HOST` = Defaults to `https://gitlab.com` but you can use it for your own url
+- `DANGER_GITLAB_API_TOKEN` = An access token for the account which will post comments
+
+Then in your Dangerfiles you will have a fully fleshed out `danger.gitlab` object to work with. For example:
+
+```ts
+import { danger, warn } from "danger"
+
+if (danger.bitbucket_cloud.mr.title.includes("WIP")) {
```
Comment (sluongng) [inline docs/usage/gitlab.html.md]: should be `danger.gitlab` instead

**Comment by orta** [comment]
Nice feedback, happy to take some PRs on it 👍

**Comment by sluongng** [comment]
> Nice feedback, happy to take some PRs on it 👍

Tried to fix it in https://github.com/danger/danger-js/pull/966

**Pull request merged**